source/javascripts/controllers/_EnvVarsController.js.erb

```diff
@@ -186,6 +199,45 @@ angular.module("BitriseWorkflowEditor").controller("EnvVarsController", function
 		return errors.length > 0 ? stringService.errorMessageFromErrors(errors) : null;
 	};
 
+	viewModel.valueVisibilityGetterSetterForEnvVar = function(secret) {
+		return function(shouldShowValue) {
+			if (shouldShowValue !== undefined) {
+				if (shouldShowValue && !secret.customProperties.progress.isInProgress) {
+					showValueOfSecret(secret);
+				}
+
+				secret.shouldShowValue = shouldShowValue;
+			}
+
+			return secret.shouldShowValue;
+		};
+	};
+
+	function showValueOfSecret(secret) {
+		secret.customProperties.progress.start("<%= data[:strings][:env_vars][:get_secret_env_value_progress][:in_progress] %>");
+
+		$q(function(resolve, reject) {
+			appService.getSecretValue(secret).then(function() {
+				resolve();
+			}, function(error) {
+				reject(error);
+			});
+		}).then(function() {
+			secret.customProperties.progress.success();
+		}, function(error) {
+			secret.customProperties.progress.error(error);
+		});
+	}
+
+	viewModel.valueGetterSetterForEnvVar = function(envVar) {
+		return function(value) {
+			return (envVar.value() !== null && envVar.shouldShowValue) ? envVar.value(value) : "******";
+		};
+	};
+
+	viewModel.makeItProtected = function(envVar) {
```

Comment:
Rename it to `makeEnvVarProtected` to indicate the subject of the action. Also, move the `.isProtected` setting inside this function. Also, don't assign it the opposite of itself, but simply a `true` value.